src/cryptoadvance/specter/server_endpoints/nodes.py

```diff
@@ -372,5 +372,18 @@ def internal_node_logs(node_alias):
 @login_required
 def switch_node():
     node_alias = request.form["node_alias"]
-    app.specter.update_active_node(node_alias)
-    return redirect(url_for("nodes_endpoint.node_settings", node_alias=node_alias))
+    node_name = app.specter.node_manager.get_name_from_alias(node_alias)
+    try:
+        app.specter.update_active_node(node_alias)
+        flash(_(f"Switched to use {node_name} as node."))
+        return redirect(url_for("index"))
+    except SpecterError:
+        flash(
+            _(f"Can't switch to {node_alias}, the node seems to not exist anymore ..."),
+            "error",
+        )
+    except Exception as e:
+        flash(
+            _(f"There's been a glitch in the matrix: {e}"),
+            "error",
+        )
```

Comment:
```
logger.exception(e)
```
please!

Reply:
Resolved in https://github.com/cryptoadvance/specter-desktop/pull/1987/commits/7e2e9a42aeb2766bdfd0a38338d814aec89358ec